packages/modules-sdk/src/medusa-app.ts

```diff
@@ -0,0 +1,179 @@
+
+  const dbData = ModulesSdkUtils.loadDatabaseConfig(
+    "medusa",
+    sharedResourcesConfig as ModuleServiceInitializeOptions,
+    true
+  )!
+  const { pool } = sharedResourcesConfig?.database ?? {}
+
+  if (dbData?.clientUrl) {
+    const { knex } = await import("@mikro-orm/knex")
+    const dbConnection = knex({
+      client: "pg",
+      searchPath: dbData.schema || "public",
+      connection: {
+        connectionString: dbData.clientUrl,
+        ssl: (dbData.driverOptions?.connection as any).ssl! ?? false,
+        idle_in_transaction_session_timeout:
+          pool?.idleInTransactionSessionTimeout ?? undefined,
+      },
+      pool: {
+        min: pool?.min ?? 0,
+        max: pool?.max,
+        idleTimeoutMillis: pool?.idleTimeoutMillis ?? undefined,
+      },
+    })
+
+    injectedDependencies[ContainerRegistrationKeys.PG_CONNECTION] = dbConnection
+  }
+
+  const allModules: Record<string, LoadedModule | LoadedModule[]> = {}
+
+  await Promise.all(
+    modules.map(async (mod: Partial<ModuleConfig> | Modules) => {
+      let key: Modules | string = mod as Modules
+      let path: string
+      let declaration: any = {}
+
+      if (isObject(mod)) {
+        if (!mod.module) {
+          throw new Error(
+            `Module ${JSON.stringify(mod)} is missing module name.`
+          )
+        }
+
+        key = mod.module
+        path = mod.path ?? MODULE_PACKAGE_NAMES[key]
+
+        declaration = { ...mod }
+        delete declaration.definition
+      } else {
+        path = MODULE_PACKAGE_NAMES[mod as Modules]
+      }
+
+      if (!path) {
+        throw new Error(`Module ${key} is missing path.`)
+      }
+
+      declaration.scope ??= MODULE_SCOPE.INTERNAL
+
+      if (
+        declaration.scope === MODULE_SCOPE.INTERNAL &&
+        !declaration.resources
+      ) {
+        declaration.resources = MODULE_RESOURCE_TYPE.SHARED
+      }
+
+      const loaded = (await MedusaModule.bootstrap(
+        key,
+        path,
+        declaration,
+        undefined,
+        injectedDependencies,
+        isObject(mod) ? mod.definition : undefined
+      )) as LoadedModule
+
+      if (allModules[key] && !Array.isArray(allModules[key])) {
+        allModules[key] = []
+      }
+
+      if (allModules[key]) {
+        ;(allModules[key] as LoadedModule[]).push(loaded[key])
+      } else {
+        allModules[key] = loaded[key]
+      }
+
+      return loaded
+    })
+  )
+
+  let link: RemoteLink | undefined = undefined
+  let query: (
+    query: string | RemoteJoinerQuery,
+    variables?: Record<string, unknown>
+  ) => Promise<any>
+
+  try {
+    const { initialize: initializeLinks } = await import(
+      "@medusajs/link-modules" as string
+    )
+    await initializeLinks({}, linkModules, injectedDependencies)
+
+    link = new RemoteLink()
+  } catch (err) {
+    console.warn("Error initializing link modules.")
+  }
+
+  const remoteQuery = new RemoteQuery(undefined, remoteFetchData)
```

Comment:
question: `undefined` here means "retrieve loaded modules from our MedusaModule class right? Which contains the modules loaded in `bootstrap` on line 126

Reply:
Exactly.